packages/iris-grid/src/IrisGridTreeTableModel.ts

```diff
@@ -44,47 +44,104 @@ class IrisGridTreeTableModel extends IrisGridTableModelTemplate<
-              isProxy: true,
-              description: 'Key column',
-              index: -1,
-              filter: () => {
-                throw new Error('Filter not implemented for virtual column');
-              },
-              sort: () => {
-                throw new Error('Sort not implemented virtual column');
-              },
-              formatColor: () => {
-                throw new Error('Color not implemented for virtual column');
-              },
-              get: () => {
-                throw new Error('get not implemented for virtual column');
-              },
-              getFormat: () => {
-                throw new Error('getFormat not implemented for virtual column');
-              },
-              formatNumber: () => {
-                throw new Error(
-                  'formatNumber not implemented for virtual column'
-                );
-              },
-              formatDate: () => {
-                throw new Error(
-                  'formatDate not implemented for virtual column'
-                );
-              },
+    this.virtualColumns = this.formatter.showExtraGroupColumn
+      ? [
+          {
+            name: '__DH_UI_GROUP__',
+            displayName: 'Group',
+            type: TableUtils.dataType.STRING,
+            constituentType: TableUtils.dataType.STRING,
+            isPartitionColumn: false,
+            isSortable: false,
+            isProxy: true,
+            description: 'Key column',
+            index: -1,
+            filter: () => {
+              throw new Error('Filter not implemented for virtual column');
             },
-          ]
-        : [];
+            sort: () => {
+              throw new Error('Sort not implemented virtual column');
+            },
+            formatColor: () => {
+              throw new Error('Color not implemented for virtual column');
+            },
+            get: () => {
+              throw new Error('get not implemented for virtual column');
+            },
+            getFormat: () => {
+              throw new Error('getFormat not implemented for virtual column');
+            },
+            formatNumber: () => {
+              throw new Error(
+                'formatNumber not implemented for virtual column'
+              );
+            },
+            formatDate: () => {
+              throw new Error('formatDate not implemented for virtual column');
+            },
+          },
+        ]
+      : [];
+  }
+
+  set formatter(newFormatter: Formatter) {
+    const oldFormatter = super.formatter;
+    super.formatter = newFormatter;
+    if (
+      newFormatter.showExtraGroupColumn !== oldFormatter.showExtraGroupColumn
+    ) {
+      this.updateVirtualColumns();
+    }
+  }
+
+  get formatter(): Formatter {
+    return super.formatter;
+  }
+
+  updateVirtualColumns(): void {
+    this.virtualColumns = this.formatter.showExtraGroupColumn
+      ? [
+          {
+            name: '__DH_UI_GROUP__',
```

Comment:
We shouldn't duplicate this object between this method and the constructor. Two options:
1. Define the constant at the top, wrap it in an `Object.freeze`, and use it in both places
2. Add a flag to `updateVirtualColumns(sendUpdate = true)` and only send an update when the param is `true`, then call `this.updateVirtualColumns(false)` from the constructor instead of assigning the property directly. Also requires adding a `!` to the `virtualColumns` declaration so that it doesn't complain about it not being initialized, e.g. `private virtualColumns!: DisplayColumn[]